File: iOS/Code/Xamarin/Insights/Insights.iOS/ViewControllers/AppsCollectionViewController.cs
using System;
using Foundation;
using UIKit;
using System.Collections.Generic;
using CoreGraphics;

namespace Insights.iOS
{
	public partial class AppsCollectionViewController : UICollectionViewController, IUICollectionViewDelegateFlowLayout
	{
		public List<MobileApp> Apps { get; set; }

		CGSize cellSize = new CGSize ((UIScreen.MainScreen.Bounds.Height > UIScreen.MainScreen.Bounds.Width) ? (UIScreen.MainScreen.Bounds.Width - 8) : (UIScreen.MainScreen.Bounds.Width - 12) / 2, 280);


		public AppsCollectionViewController (IntPtr handle) : base(handle)
		{
			Apps = InsightsService.GetApps;
		}


		public override nint NumberOfSections (UICollectionView collectionView)
		{
			return 1;
		}


		public override nint GetItemsCount (UICollectionView collectionView, nint section)
		{
			return Apps.Count;
		}


		public override UICollectionViewCell GetCell (UICollectionView collectionView, NSIndexPath indexPath)
		{
			var cell = collectionView.DequeueReusableCell(AppsCollectionViewCell.ReuseId, indexPath) as AppsCollectionViewCell;

			var app = Apps[Convert.ToInt32(indexPath.Item)];

			cell.SetContent(cellSize, app);

			return cell;
		}


		public override void ItemSelected (UICollectionView collectionView, NSIndexPath indexPath)
		{
			var tabController = Storyboard.Instantiate<AppTabBarController>();
			ShowViewController(tabController, this);
		}


		[Export("collectionView:layout:sizeForItemAtIndexPath:")]
		public CGSize GetSizeForItem (UICollectionView collectionView, UICollectionViewLayout layout, NSIndexPath indexPath)
		{
			return cellSize;
		}


		public override UIStatusBarStyle PreferredStatusBarStyle ()
		{
			return UIStatusBarStyle.LightContent;
		}
	}
}

File: iOS/Code/Swift/bugTrap/SampleHostApp/Controllers/AppsCollectionViewController.swift
//
//  AppsCollectionViewController.swift
//  SampleHostApp
//
//  Created by Colby Williams on 8/22/15.
//  Copyright © 2015 bugTrap. All rights reserved.
//


import Foundation;
import UIKit;
import CoreGraphics;

class AppsCollectionViewController : UICollectionViewController, UICollectionViewDelegateFlowLayout
{
	
	let apps = [ "BloodPressureX", "bugTrap", "bugTrap Debug", "VervetaCRM.Android", "VervetaCRM.iOS", "Glucose X" ]
	
	
	override func numberOfSectionsInCollectionView(collectionView: UICollectionView) -> Int {
		return 1
	}
	
	override func collectionView(collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
		return apps.count
	}
	
	override func collectionView(collectionView: UICollectionView, cellForItemAtIndexPath indexPath: NSIndexPath) -> UICollectionViewCell {
		
		let cell = collectionView.dequeueReusableCellWithReuseIdentifier("AppsCollectionViewCell", forIndexPath: indexPath) as! AppsCollectionViewCell

		let app = apps[indexPath.row]
		
		cell.setContent(cellSize, app: app)
		
		return cell
	}
	
	override func collectionView(collectionView: UICollectionView, didSelectItemAtIndexPath indexPath: NSIndexPath) {
		if let tabController = storyboard?.instantiateViewControllerWithIdentifier("AppTabBarController") {
			showViewController(tabController, sender: self)
		}
	}

	
	override func preferredStatusBarStyle() -> UIStatusBarStyle {
		return .LightContent
	}

	let cellSize = CGSize(width: UIScreen.mainScreen().bounds.height > UIScreen.mainScreen().bounds.width ? UIScreen.mainScreen().bounds.width - 8 : (UIScreen.mainScreen().bounds.width - 12) / 2, height: 280)
	
	func collectionView(collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAtIndexPath indexPath: NSIndexPath) -> CGSize {
		return cellSize
	}
}

